### PR TITLE
CVE-2010-1452

### PR DESCRIPTION
**Pull request by JMdoubleU**

### CODE REVIEW

File: cves/CVE-2010-1452.yml [hunk 1]
```diff
@@ -204,4 +212,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This issue seems attributable to a coding mistake, as the implementation
+    of request parsing made an assumption.
```
Comment (mesternefeld) [inline cves/CVE-2010-1452.yml]: Maybe elaborate more on this. It seems a little short.

File: cves/CVE-2010-1452.yml [hunk 2]
```diff
@@ -80,9 +85,9 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer:
-  code:
-  fix:
+  answer: No unit tests existed for the affected modules.
```
Comment (cxd3796) [inline cves/CVE-2010-1452.yml]: Were there different types of tests or any types of tests at all? For example were there integration tests that could have mitigated the problem?

File: cves/CVE-2010-1452.yml [hunk 3]
```diff
@@ -122,6 +128,8 @@ interesting_commits:
     Write a brief (under 100 words) description of why you think this commit was
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
+  answer: |
+    There are no especially interesting commits between the VCCs and fix commits.
```
Comment (cxd3796) [inline cves/CVE-2010-1452.yml]: Was there anything between the VCCs and the fix commits that may have led a developer to the fixes or potentially to discover a different problem or vulnerability?

File: cves/CVE-2010-1452.yml [hunk 4]
```diff
@@ -171,8 +179,8 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
```
Comment (cxd3796) [inline cves/CVE-2010-1452.yml:184]: Is this the only lesson application? it sounds like there could be potential for other ones.

Comment (JMdoubleU) [reply cves/CVE-2010-1452.yml:184]: I'm not sure anything else can be applied. This was a simple case of forgetting to check if a pointer was `NULL`. Any suggestions?

File: cves/CVE-2010-1452.yml (hunk 2 above)
Comment (JMdoubleU) [reply cves/CVE-2010-1452.yml]: There were informal test scripts in the repo, but they did not touch the affected modules. Will update to reflect that.

File: cves/CVE-2010-1452.yml (hunk 3 above)
Comment (JMdoubleU) [reply cves/CVE-2010-1452.yml]: I found that another somewhat similar vulnerability was fixed in this time period. Will update with the details.

File: cves/CVE-2010-1452.yml (hunk 1 above)
Comment (JMdoubleU) [reply cves/CVE-2010-1452.yml]: Expanded on it a little bit - didn't want to get too technical.

**Pull request merged**